pkg/cache/application.go

```diff
@@ -166,9 +167,16 @@ func (app *Application) GetSchedulingParamsDefinition() string {
 func (app *Application) setTaskGroups(taskGroups []TaskGroup) {
 	app.lock.Lock()
 	defer app.lock.Unlock()
-	app.taskGroups = taskGroups
-	for _, taskGroup := range app.taskGroups {
-		app.placeholderAsk = common.Add(app.placeholderAsk, common.GetTGResource(taskGroup.MinResource, int64(taskGroup.MinMember)))
+	for _, tg := range taskGroups {
+		taskGroup := tg
```

Comment:
If the map is `string[TaskGroup]` then this is not needed I believe.

Reply:
> If the map is `string[TaskGroup]` then this is not needed I believe.

Yes, if the map is `string[TaskGroup]`, it really doesn 't need to be done here.